brontide/noise.go

```diff
@@ -703,15 +703,30 @@ func (b *Machine) WriteMessage(w io.Writer, p []byte) error {
 
 	// First, write out the encrypted+MAC'd length prefix for the packet.
 	cipherLen := b.sendCipher.Encrypt(nil, nil, pktLen[:])
-	if _, err := w.Write(cipherLen); err != nil {
+	if err := safeWrite(w, cipherLen); err != nil {
 		return err
 	}
 
 	// Finally, write out the encrypted packet itself. We only write out a
 	// single packet, as any fragmentation should have taken place at a
 	// higher level.
 	cipherText := b.sendCipher.Encrypt(nil, nil, p)
-	_, err := w.Write(cipherText)
+	return safeWrite(w, cipherText)
+}
+
+// safeWrite writes the message p to the given io.Writer. An error is returned
+// if the message was partially written, i.e. the number of bytes written was
+// in the range [1, len(p)). This allows us to propagate timeout errors that
```

Comment:
Ah, so for `n == 0` we want to return the error unaltered?

Reply:
Yeah because we never wrote anything, so it was most likely a timeout or EOF.